sdk/dslink/src/main/java/org/dsa/iot/dslink/link/SubscriptionHelper.java

```diff
@@ -0,0 +1,192 @@
+package org.dsa.iot.dslink.link;
+
+import java.util.Comparator;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.dsa.iot.dslink.node.value.SubscriptionValue;
+import org.dsa.iot.dslink.util.SubData;
+import org.dsa.iot.dslink.util.handler.Handler;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * The requester api is written in such a way that multiple subscriptions to the same
+ * path are not possible.  This uses a multiplexing adapter so that the requester
+ * thinks there is only one subscriber.
+ *
+ * @author Aaron Hansen
+ */
+public class SubscriptionHelper {
+
+    // Fields
+    // ------
+
+    private Requester requester;
+    private ConcurrentHashMap<String, Adapter> subscriptions;
+
+    // Constructors
+    // ------------
+
+    public SubscriptionHelper(Requester requester) {
+        if (requester == null) {
+            throw new NullPointerException("Requester must not be null");
+        }
+        this.requester = requester;
+        subscriptions = new ConcurrentHashMap<String, Adapter>();
+    }
+
+
+    // Public Methods
+    // --------------
+
+    /**
+     * Clears all subscriptions, but does not call unsubscribe on the requester.
+     */
+    public void clear() {
+        if (subscriptions != null) {
+            subscriptions.clear();
+        }
+    }
+
+    /**
+     * Clears a single subscription without calling unsubscribe on the requester.
+     */
+    public SubscriptionHelper clear(String path) {
+        Adapter a = subscriptions.remove(path);
+        if (a != null) {
+            requester.unsubscribe(path, null);
+        }
+        return this;
```

Comment:
I'm the opposite, I'm a huge fan of being able to chain calls.  I do it out of habit.